app/assets/javascripts/models/vacation_request.js

```diff
@@ -1,3 +1,29 @@
 App.Models.VacationRequest = Backbone.Model.extend({
-  urlRoot: 'vacation_requests'
+  urlRoot: 'vacation_requests',
+  defaults: {
+    'end':'',
+    'duration':1,
+    'kind':'planned',
+    'start':'',
+    'status':'requested',
+  },
+
+  // Calculate duration and set duration attribute with the result
+  // Solution is based on sets of days and their unions and instersections.
+  calculateDuration: function(holidays, options) {
+    var result = 0,
+        arrayOfHolidays = [],
+        arrayOfVacationDays = [],
+        arrayOfWeekends = [],
+        numberOfDays = moment.duration(moment(this.get('end')).diff(moment(this.get('start')),'days'))+1;
```

Comment:
Let's leave it alone for a while: this model is to be reworked due to DB schema changes.